src/test/java/org/mskcc/cbio/oncokb/web/rest/UsageAnalysisControllerIT.java

```diff
@@ -0,0 +1,931 @@
+        String[] userUsageSummaryEndpoints = {
+          "/api/usage/summary/users",
+          "/api/usage/summary/users?companyId=" + companyDTO.getId(),
+        };
+
+        for (int yearIndex = 0; yearIndex < 4; yearIndex++) {
+          int year = currentYear - yearIndex;
+          Boolean isThisYear = yearIndex == 0;
+
+          DateTimeFormatter dayFormatter = DateTimeFormatter.ofPattern(
+            "yyyy-MM-dd"
+          );
+          DateTimeFormatter monthFormatter = DateTimeFormatter.ofPattern(
+            "yyyy-MM"
+          );
+
+          String yearKey = String.valueOf(year);
+
+          for (int month = 1; month <= 12; month++) {
+            YearMonth yearMonth = YearMonth.of(year, month);
+            String monthKey = yearMonth.format(monthFormatter);
+            LocalDate start = yearMonth.atDay(1).atStartOfDay().toLocalDate();
+            LocalDate end = yearMonth
+              .atEndOfMonth()
+              .atStartOfDay()
+              .toLocalDate();
+
+            for (
+              LocalDate date = start;
+              !date.isAfter(end) && !date.isAfter(today);
+              date = date.plusDays(1)
+            ) {
+              int j = 0;
+              String dayKey = date.format(dayFormatter);
+              while (random.nextInt() % 5 == 0 && j < 5) {
+                j++;
+                String oncokbEndpoint = getNextOncokbEndpoint();
+                int value = random.nextInt(100);
+
+                updateResourceYearFile(
+                  value,
+                  oncokbEndpoint,
+                  yearKey,
+                  monthKey
+                );
+
+                updateResourceMonthFile(
+                  value,
+                  oncokbEndpoint,
+                  monthKey,
+                  dayKey
+                );
+
+                updateUserYearFile(
+                  value,
+                  user,
+                  oncokbEndpoint,
+                  yearKey,
+                  monthKey
+                );
+
+                updateUserMonthFile(
+                  value,
+                  user,
+                  oncokbEndpoint,
+                  monthKey,
+                  dayKey
+                );
+
+                boolean isAfterOrEqualToPast12Months = date.isAfter(
+                  today.minusMonths(11).withDayOfMonth(1).minusDays(1)
+                );
+                boolean isBeforeOrEqualToToday = date.isBefore(
+                  today.plusMonths(1).withDayOfMonth(1)
+                );
+
+                // update expected response for monthly resource summary
+                if (isThisYear) {
+                  updateResourceSummaryExpectedResponse(
+                    value,
+                    oncokbEndpoint,
+                    monthKey
+                  );
+                }
+
+                // update expected response for user usage endpoint
+                if (isThisYear) {
```

Comment:
@zhx828 I know limiting the usage metrics to only "this" year isn't what we want, but it is currently how the code works.

Reply:
You expect to change this after finishing "3 years" requirement to make the test pass?

Reply:
Yes